### PR TITLE
Cannot read properties of null (reading 'id')

### PR DESCRIPTION
**Pull request by qq282126990**
If columnKey is not set, execute clearFilter and prompt Cannot read properties of null (reading 'id')
Suggestion: Add judgment to clearly prompt developers

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by qq282126990** [comment]
example: image

**Comment by btea** [comment]
@qq282126990 Hello, please provide a recurrence link

**Comment by qq282126990** [comment]
Hi please check here
 https://github.com/qq282126990/element-plus/blob/dev/packages/components/table/src/util.ts#L120 
 @btea

**Comment by btea** [comment]
Ok, I see.

**Comment by qq282126990** [comment]
@btea  Hi I changed the format of the code and it should now pass the test 
image

**Comment by qq282126990** [comment]
@btea hello please try the test case
image

**Comment by qq282126990** [comment]
@btea  hello please try the test case
image

**Comment by btea** [comment]
In this case, is it acceptable to throw an error directly, what do you think? @element-plus/backers

File: packages/components/table/src/util.ts [hunk 1]
```diff
@@ -115,6 +115,7 @@ export const getColumnByKey = function <T>(
   let column = null
   for (let i = 0; i < table.columns.length; i++) {
     const item = table.columns[i]
+    if (!item.columnKey) throw new Error('column does not have columnKey set')
```
Comment (holazz) [inline packages/components/table/src/util.ts:118]:
```suggestion
```

```diff
    for (let i = 0; i < table.columns.length; i++) {
      const item = table.columns[i]
-     if (!item.columnKey) throw new Error('column does not have columnKey set')
      if (item.columnKey === columnKey) {
        column = item
        break
      }
    }
+   if (!column) throw new Error(`[Table] No column matching with column-key: ${columnKey}`)
    return column
```

**Review by holazz — COMMENTED** [review]
I think the error should be thrown after all columns are checked and there is no match. The error message should be something like "No matching column". Because we may have set the `column-key`, but the wrong key was passed in when calling the `clearFilter` method, eg:

```vue
<el-table-column>
  column-key="date"
  ...
</el-table-column>
```

```ts
tableRef.value.clearFilter(['data']) // should be 'date'
```

File: packages/components/table/src/util.ts (hunk 1 above)
Comment (btea) [reply packages/components/table/src/util.ts:118]: good suggestion 👍 !  Maybe also use `throwError`

https://github.com/element-plus/element-plus/blob/742a37ca58aa4833653b3c0d91d5157610ce4463/packages/utils/error.ts#L10

Comment (holazz) [reply packages/components/table/src/util.ts:118]: That's much better！I didn't notice it before.

**Comment by btea** [comment]
@qq282126990 Hi, can you modify the code as suggested by @holazz ? And please follow the [prompts](https://github.com/element-plus/element-plus/pull/9322#issuecomment-1219074731) to modify the submission information.

**Comment by btea** [comment]
The test seems to have failed.

**Comment by qq282126990** [comment]
@btea  The test case cannot find the file
image
image

**Comment by btea** [comment]
@qq282126990 It seems that there is a problem with the file path, maybe you can try to create a new pr.
image

**Comment by qq282126990** [comment]
@btea Hi I recreated a new pull to go  
https://github.com/element-plus/element-plus/pull/9682

**Pull request closed without merging**